src/languages/asciidoc.js

```diff
@@ -110,30 +141,47 @@ export default function(hljs) {
       // inline unconstrained strong (single line)
       {
         className: 'strong',
-        begin: /\*\*(.+?)\*\*/
+        begin: `\\*{2}(${CC_ANY}+?)\\*{2}`
       },
       // inline unconstrained strong (multi-line)
       {
         className: 'strong',
-        begin: /\*{2}(\s\S+)*\*{2}/,
+        begin: /\*{2}(([^*\n\\]|\\[^\n])+\n)+([^*\n\\]|\\[^\n])*\*{2}/,
         relevance: 0
       },
-      // escaped formatting marks (i.e., \* \_ or \`)
+      // escaped constrained formatting marks (i.e., \* \_ or \`)
       {
         begin: /\\[*_`]/
       },
-      // inline contrained string (single line)
+      // escaped unconstrained formatting marks (i.e., \\** \\__ or \\``)
+      // must ignore until the next formatting marks
+      // this rule is not 100% compliant with Asciidoctor but we are entering undefined behavior territory...
+      {
+        begin: /\\\\\*{2}[^\n]*\*{2}/
```

Comment:
Wait does this syntax literally mean "it's marked as bold, but ignore it and don't make it bold"?

Reply:
Also shouldn't this be CC_ANY?

Reply:
Oh and you don't need to turn them all into strings either, just use `regex.concat`.

Reply:
> Wait does this syntax literally mean "it's marked as bold, but ignore it and don't make it bold"?

It's another guard when unconstrained bold is escaped using `\\`. Take a look at the test case:

```
\\**--anything goes **
```

I've also added an even more complex test case:

```
\\**--anything goes ** with * a **lot** of stars**
```

> Also shouldn't this be CC_ANY?

Yes

> Oh and you don't need to turn them all into strings either, just use regex.concat.

Oh really? Could you please provide an example, I've never seen `regex.concat` :exploding_head:

Reply:
It's in `lib/regex.js`...  just utility functions.

Reply:
> It's another guard when unconstrained bold is escaped using \\. Take a look at the test case: 

OK if really it's like a paired "don't apply any highlighting on the inside", ie "anything goes"... (kind of like a comment in a programming language) but why are there 3 variants of it? :-)  Asciidoc is strange. :)